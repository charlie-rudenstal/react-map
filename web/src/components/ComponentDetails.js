import React from 'react';

class ComponentDetails extends React.Component {
  constructor(props) {
    super(props);
    this.state = { component: null };
    this.fetchComponent = this.fetchComponent.bind(this);
  }

  componentDidMount() {
    this.fetchComponent(this.props.path);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.path === this.props.path) return;
    this.fetchComponent(nextProps.path);
  }

  fetchComponent(path) {
    fetch(`/api/component/${path}`)
      .then(res => {
        if (!res.ok) throw Error(`Could not fetch component ${this.props.path}`);
        return res.json();
      })
      .catch(error => this.setState({error}))
      .then(component => this.setState({component}));
  }

  render() {
    const { component, error } = this.state;
    if (error) return <div>{error.toString()}</div>;
    if (!component) return <div>Fetching component data...</div>;

    return (
      <div>
        <h1>{component.name}</h1>

        <section>
          <h2>Children</h2>
          <ul>
          {component.children.map(item => (
            <li key={item.name}>{item.name}</li>
          ))}
          </ul>
        </section>

        <section>
          <h2>Class names</h2>
          <ul>
          {component.classNames.map(item => (
            <li key={item.name}>{item.name}</li>
          ))}
          </ul>
        </section>

        <section>
          <h2>Dependencies</h2>
          <ul>
          {component.dependencies.map(item => (
            <li key={item.name}>{item.name} ({item.path})</li>
          ))}
          </ul>
        </section>
      </div>
    );
  }
}

ComponentDetails.propTypes = {
  path: React.PropTypes.string.isRequired
};

export default ComponentDetails;
