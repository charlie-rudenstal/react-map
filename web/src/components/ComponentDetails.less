:local(.content) {
  margin: 12px;
}