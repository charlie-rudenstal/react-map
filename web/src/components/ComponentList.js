import React from 'react';

class ComponentList extends React.Component {

  constructor(props) {
    super(props);
    this.state = {components: []};
  }

  componentDidMount() {
    fetch('/api/components')
      .then(res => res.json())
      .then(({components}) => {
        this.setState({components});
      });
  }

  render() {
    return (
      <div>
        <div>ComponentList</div>
        <ul>
          {this.state.components.map(component => (
            <div key={component.name}>
              {component.name}
            </div>
          ))}
        </ul>
      </div>
    );
  }

}

export default ComponentList;