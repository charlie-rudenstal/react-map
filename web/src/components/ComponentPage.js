import React from 'react';

function ComponentPage() {
  return (
    <div>Component Page</div>
  );
}

export default ComponentPage;