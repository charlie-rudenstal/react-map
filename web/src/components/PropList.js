import React from 'react';
import provideApi from '../lib/provideApi';
import { withRouter } from 'react-router';

function PropList({component, router}) {
  if (!component) return null;
  console.log('component is', component);
  return (
    <div className="itemlist">
      {component.props.map(({name, type}) => (
        <div className="itemlist__item">
          {name}
        </div>
      ))}
    </div>
  );
}

export default withRouter(provideApi(PropList,
  props => `/component/${props.router.params.splat}`,
  component => ({ component })
));
