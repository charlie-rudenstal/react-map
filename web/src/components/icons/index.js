export {default as FileIcon} from './File';
