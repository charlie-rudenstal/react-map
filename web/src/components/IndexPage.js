import React from 'react';
import ComponentList from './ComponentList';

function IndexPage() {
  return (
    <div>
      <ComponentList />
    </div>
  );
}

export default IndexPage;
