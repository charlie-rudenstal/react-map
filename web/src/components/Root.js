import React from 'react';
import ComponentList from './ComponentList';

function Root() {
  return (
    <div>
      <ComponentList />
    </div>
  );
}

export default Root;
