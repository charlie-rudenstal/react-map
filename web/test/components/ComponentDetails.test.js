import React from 'react';
import renderer from 'react-test-renderer';
import { ComponentDetails } from '../../src/components/ComponentDetails';

describe('ComponentDetails', () => {
  it('should render correctly', () => {
    const component = {
      name: "Tabs",
      dependencies: [
        { name: "React", path: "react" },
        { name: "Tab", path: "./Tab" }
      ],
      children: [
        { name: "div" },
        { name: "Tab" }
      ],
      classNames: [
        { name: "tabs" },
        { name: "header" },
        { name: "header--large" }
      ]
    };

    const tree = renderer.create(<ComponentDetails path='/component/tabs' component={component} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
