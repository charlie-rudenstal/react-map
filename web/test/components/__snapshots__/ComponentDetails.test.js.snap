exports[`ComponentDetails should render correctly 1`] = `
<div>
  <h1>
    Tabs
  </h1>
  <section>
    <h2>
      Children
    </h2>
    <ul>
      <li>
        div
      </li>
      <li>
        Tab
      </li>
    </ul>
  </section>
  <section>
    <h2>
      Class names
    </h2>
    <ul>
      <li>
        tabs
      </li>
      <li>
        header
      </li>
      <li>
        header--large
      </li>
    </ul>
  </section>
  <section>
    <h2>
      Dependencies
    </h2>
    <ul>
      <li>
        React
         (
        react
        )
      </li>
      <li>
        Tab
         (
        ./Tab
        )
      </li>
    </ul>
  </section>
</div>
`;
